packages/neos-ui/src/localStorageMiddleware.js

```diff
@@ -0,0 +1,41 @@
+import {$get} from 'plow-js';
+
+//
+// Local storage middleware.
+//
+// Saves "ui" part of the state to local storage on every action
+// that matches the "persistentActionsPattern"
+//
+const localStorageMiddleware = ({getState}) => {
+    let timer = null;
+    const debounceLocalStorageTimeout = 1000;
+    const persistentActionsPatterns = ['@neos/neos-ui/UI', '@neos/neos-ui/User/Settings'];
```

Comment:
This feels a bit wacky to me now, I think we need a more expressive convention of how we decide what to store in the local storage and what not.